wagtailio/core/blocks.py

```diff
@@ -374,3 +374,165 @@ class IconBulletsBlock(blocks.StructBlock):
     class Meta:
         # template = "" # TODO: add template
         label = "Icon bullets"
+
+
+class StandaloneQuoteBlock(blocks.StructBlock):
+    quote = blocks.TextBlock(required=True)
+    author = blocks.RichTextBlock(
+        required=True,
+        features=[
+            "bold",
+            "italic",
+            "h2",
+            "h3",
+            "h4",
+            "h5",
+            "ol",
+            "ul",
+            "link",
+            "image",
+            "document",
+            "embed",
+        ],
+    )
+    author_image = ImageChooserBlock(required=False)
+
+    class Meta:
+        icon = "openquote"
+        label = "Standalone quote"
+        template = "patterns/components/streamfield/standalone_quote_block.html"
+
+
+class MultipleQuoteBlock(blocks.StructBlock):
+    heading = blocks.TextBlock(required=True)
+    quotes = blocks.ListBlock(StandaloneQuoteBlock(), min_num=2)
+
+    class Meta:
+        icon = "openquote"
+        label = "Multiple quotes"
+        template = "patterns/components/streamfield/multiple_quotes_block.html"
+
+
+class RichTextBlock(blocks.StructBlock):
+    rich_text = blocks.RichTextBlock(
+        required=True,
+        features=["bold", "italic", "h2", "h3", "ol", "ul", "link", "document"],
+    )
+
+    class Meta:
+        icon = "bold"
+        label = "Rich text"
+        template = "patterns/components/streamfield/rich_text_block.html"
+
+class StandaloneCTABlock(blocks.StructBlock):
+    short_description = blocks.TextBlock(required=False, max_length=100)
+    cta = CTABlock()
+
+    class Meta:
+        icon = "arrow-right"
+        label = "Standalone CTA"
+        template = "patterns/components/streamfield/standalone_cta_block.html"
+
+
+class TeaserBlock(blocks.StructBlock):
+    blog_post_chooser = blocks.PageChooserBlock(required=False, page_type=["blog.BlogPage"])
+    url_chooser = blocks.URLBlock(required=False)
+    image_for_external_link = ImageChooserBlock(required=False)
+    heading_for_external_link = blocks.TextBlock(required=False)
+    subheading_for_ext_link = blocks.TextBlock(
+        label="Subheading for external link", required=False
+    )
+
+    def clean(self, value):
+        errors = {}
+        struct_value = super().clean(value)
+
+        # We want the logical AND of three values
+        sum_external_link_values = (
+            bool(value.get("image_for_external_link"))
+            + bool(value.get("heading_for_external_link"))
+            + bool(value.get("subheading_for_ext_link"))
+        )
+
+        if not value.get("blog_post_chooser") and not value.get("url_chooser"):
+            e = ErrorList(
```

Comment:
nitpick: can this be called error. Can't shake the `Except as e` usage out of my head :)

Reply:
done in f3751fde600de09043686218a674a322b0039046 :)